draft-ietf-quic-recovery.md

```diff
@@ -762,47 +762,95 @@ The minimum congestion window is the smallest value the congestion window can
 decrease to as a response to loss, ECN-CE, or persistent congestion.
 The RECOMMENDED value is 2 * max_datagram_size.
 
-## Slow Start
+## Congestion Control States
 
-While in slow start, QUIC increases the congestion window by the
-number of bytes acknowledged when each acknowledgment is processed, resulting
-in exponential growth of the congestion window.
+The congestion controller has three distinct states, as shown in
+{{fig-cc-fsm}}.
 
-QUIC exits slow start upon loss or upon increase in the ECN-CE counter.
-When slow start is exited, the congestion window halves and the slow start
-threshold is set to the new congestion window.  QUIC re-enters slow start
-any time the congestion window is less than the slow start threshold,
-which only occurs after persistent congestion is declared.
+~~~
+                 Start or         +------------+
+            persistent congestion |   Slow     |
+        (O)---------------------->|   Start    |
+                                  +------------+
+                                        |
+                                Loss or |
+                        ECN-CE increase |
+                                        v
```

Comment:
```suggestion
            Start                 +------------+
             (O)----------------->|   Slow     |
       +------------------------->|   Start    |
       |                          +------------+
       | Persistent                     |
       | congestion             Loss or |
       |                ECN-CE increase |
       |                                v
```

It's probably worth calling persistent congestion out as a separate transition rather than a restart.  Though it's more complex than that, since I believe persistent congestion is also checked only on a loss event.

Reply:
You could enter persistent congestion from any state, so this particular transition won't be adequate. You could make draw a parallel box called persistent-congestion on the left to which {Slow Start, Congestion Avoidance, Recovery} can all transition, which would then transition back up to Slow Start.

Reply:
When you encounter persistent congestion, you reset the state machine.  That is what the text below the figure says.  It's not a new state because then you are in Slow Start again.  I think that is the only sensible solution here.

Reply:
Good point. It's certainly not a new state for the congestion controller; it's an event, much like loss.